src/wechaty_plugin_contrib/contrib/chat_history_plugin/plugin.py

```diff
@@ -56,38 +57,48 @@ class ChatHistoryPlugin(WechatyPlugin):
 
     def __init__(self, options: Optional[ChatHistoryPluginOptions] = None):
         super().__init__(options)
+        if options is None:
+            options = ChatHistoryPluginOptions()
         if not options.chat_history_path:
             self.chat_history_path = os.path.join(os.getcwd(), 'chathistory')
         if not os.path.exists(self.chat_history_path):
             os.makedirs(self.chat_history_path)
-        if not options.chat_history_database:
-            self.chat_history_database = 'sqlite+aiosqlite:///chathistory.db'
-        else:
-            self.chat_history_database = options.chat_history_database
+        self.chat_history_database = options.chat_history_database \
+            or 'sqlite+aiosqlite:///chathistory.db'
 
     @property
     def name(self) -> str:
         return 'chat-history'
 
-    async def on_message(self, msg: Message):
-        """listen message event"""
+    async def init_plugin(self, wechaty: Wechaty):
+        """init plugin"""
         async_engine = create_async_engine(self.chat_history_database)
         async with async_engine.begin() as conn:
             await conn.run_sync(Base.metadata.create_all)
+
+    async def on_message(self, msg: Message):
+        """listen message event"""
+        async_engine = create_async_engine(self.chat_history_database)
         async_session = sessionmaker(async_engine,
                                      expire_on_commit=False,
                                      class_=AsyncSession)
         async with async_session() as session:
             async with session.begin():
+                file_box: FileBox = None
+                if msg.type() in SUPPORTED_MESSAGE_FILE_TYPES:
+                    file_box = await msg.to_file_box()
```

Comment:
You only save the filename, not file content which is more important. I suggest that you can save it as the local file, and save the file-path to database. How do you think about it ?

Reply:
Yes, I save the filename to database and save the file content to the `chathistory` folder(optional path). But if we want to get the filename and content, we need to load the palyload stream twice (one for filename and another for the content because of async).